sql/src/main/java/io/crate/metadata/DanglingIndex.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.metadata;
+
+
+public class DanglingIndex {
+    private static final String SHRINK_KEYWORD = "shrinked";
+    private static final String BACKUP_KEYWORD = "backup";
+
+    public static final String SHRINK_PREFIX = "." + SHRINK_KEYWORD + ".";
+    public static final String BACKUP_PREFIX = "." + BACKUP_KEYWORD + ".";
+
+    private DanglingIndex() {}
+
+    private static boolean isShrinkedIndex(String indexName) {
+        return indexName.startsWith(SHRINK_PREFIX);
+    }
+
+    private static boolean isBackupIndex(String indexName) {
+        return indexName.startsWith(BACKUP_PREFIX);
+    }
+
+    public static boolean isDanglingIndex(String indexName) {
+        return isShrinkedIndex(indexName) || isBackupIndex(indexName);
```

Comment:
I'd prefer doing it inverse, like filter everything starting with a `.` beside `IndexParts.PARTITIONED_TABLE_PART`

Reply:
@mfussenegger @andreidan what's your opinion on that?

Reply:
I like this proposal @seut , pushed changes